src/test/java/com/instana/operator/AgentDeployerTest.java

```diff
@@ -70,20 +78,75 @@ void daemonset_must_include_specified_image() {
   @Test
   void daemonset_must_include_image_from_csv_if_specified() {
     AgentDeployer deployer = new AgentDeployer();
+    deployer.setDefaultClient(client);
     deployer.setEnvironment(singleVar(RELATED_IMAGE_INSTANA_AGENT, "other/image:some-tag"));
 
-    InstanaAgent crd = new InstanaAgent();
-    crd.setSpec(new InstanaAgentSpec());
+    InstanaAgent customResource = new InstanaAgent();
+    customResource.setSpec(new InstanaAgentSpec());
 
     DaemonSet daemonSet = deployer.newDaemonSet(
-        crd,
+        customResource,
         client.inNamespace("instana-agent").apps().daemonSets());
 
     Container agentContainer = getAgentContainer(daemonSet);
 
     assertThat(agentContainer.getImage(), is("other/image:some-tag"));
   }
 
+  @Test
+  public void daemonset_must_include_version_label_if_specified_on_crd() {
+    // create crd from yaml
+    client.load(getClass().getResourceAsStream("/customresourcedefinition-with-version.yaml"))
+        .createOrReplace()
+        .stream()
+        .findAny()
+        .orElse(null);
```

Comment:
Does this rely on having some kube context present? I don't the rest of these tests relied on this, which may be why they are failing on jenkins / circleci.

Reply:
Yeah. My mistake. Fixing now.